src/Adapter/OrderReturn/CommandHandler/UpdateOrderReturnStateHandler.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\OrderReturn\CommandHandler;
+
+use OrderReturn;
+use PrestaShop\PrestaShop\Adapter\OrderReturn\AbstractOrderReturnHandler;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\Command\UpdateOrderReturnStateCommand;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\CommandHandler\UpdateOrderReturnStateHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\Exception\OrderReturnException;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\Exception\OrderReturnOrderStateConstraintException;
+
+class UpdateOrderReturnStateHandler extends AbstractOrderReturnHandler implements UpdateOrderReturnStateHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(UpdateOrderReturnStateCommand $command): void
+    {
+        $orderReturnId = $command->getOrderReturnId();
+
+        $orderReturn = $this->getOrderReturn($orderReturnId);
+
+        $orderReturn = $this->updateOrderReturnWithCommandData($orderReturn, $command);
+
+        $this->assertRequiredFieldsAreNotMissing($orderReturn);
+
+        if (false === $orderReturn->validateFields(false)) {
+            throw new OrderReturnException('Order return contains invalid field values');
```

Comment:
Could you add behat tests for this behavior ?

Reply:
Not really.

It's impossible to get here with the current code structure. Only state is updated, and it will always be int due to strict types. Any int will always pass validate fields for state.

This here just in case something goes really wrong in the code somewhere else and affects this handler.

I think I could add unit test, where updateOrderReturnWithCommandData is mocked to alert state and make it incorrect. What do you think?

Reply:
Validation could be improved by validating each field using legacy object model validator (as sone in Product subdomains). But this doesnt block the pr i guess, it could be done in separate one